javascripts/discourse/services/homepage-filter.js

```diff
@@ -4,86 +4,120 @@ import Service, { service } from "@ember/service";
 import { ajax } from "discourse/lib/ajax";
 import Category from "discourse/models/category";
 
-const count = 50; // TODO: turn this into a loadmore component
-
 export default class HomepageFilter extends Service {
   @service siteSettings;
 
   @tracked tagFilter = null;
   @tracked topicResults = [];
   @tracked searchQuery = "";
   @tracked inputText = "";
-  @tracked loading = true;
+  @tracked loading = false;
+  @tracked hasMoreResults = false;
+  @tracked currentPage = 1;
 
   updateFilter(filter) {
     if (this.tagFilter !== filter) {
       this.tagFilter = filter;
+      this.resetPageAndFetch();
     }
   }
 
-  get filter() {
-    if (this.searchQuery) {
-      const category = Category.findById(
-        parseInt(this.siteSettings.discourse_discover_category, 10)
-      );
+  updateSearchQuery(query) {
+    if (this.searchQuery !== query) {
+      this.searchQuery = query;
+      this.resetPageAndFetch();
+    }
+  }
+
+  resetPageAndFetch() {
+    this.currentPage = 1;
+    this.topicResults = [];
+    this.hasMoreResults = false;
+    this.getSiteList();
+  }
+
+  createSrcset(thumbnails) {
+    return thumbnails
+      .map((thumbnail) => `${thumbnail.url} ${thumbnail.width}w`)
+      .join(", ");
+  }
+
+  createBannerImage(thumbnails) {
+    const srcset = this.createSrcset(thumbnails);
+    const sizes = "(max-width: 768px) 100vw, (max-width: 1200px) 50vw, 33vw";
+    const src = thumbnails[0].url;
+
+    return { srcset, sizes, src };
+  }
+
+  parseResults(results) {
+    return results.topics.map((topic) => ({
+      ...topic,
+      discover_entry: topic.discover_entry
+        ? {
+            ...topic.discover_entry,
+            extra: topic.discover_entry.extra
+              ? JSON.parse(topic.discover_entry.extra)
+              : {},
+          }
+        : {},
+      bannerImage: topic.thumbnails
+        ? this.createBannerImage(topic.thumbnails)
+        : null,
+    }));
+  }
 
-      const tagFilter = this.tagFilter ? ` tags%3A${this.tagFilter}` : "";
+  get currentFilter() {
+    const category = Category.findById(
+      parseInt(this.siteSettings.discourse_discover_category, 10)
+    );
 
-      return `/search.json?q=${this.searchQuery} %23${category.slug}${tagFilter}`;
-    } else {
-      return this.tagFilter
-        ? `/tags/c/${this.siteSettings.discourse_discover_category}/${this.tagFilter}.json`
-        : `c/${this.siteSettings.discourse_discover_category}.json`;
+    let searchString = `#${category.slug}`;
+    if (this.tagFilter) {
+      searchString += ` tags:${this.tagFilter}`;
+    }
+    if (this.searchQuery) {
+      searchString += ` ${this.searchQuery}`;
     }
+
+    return `/search.json?q=${encodeURIComponent(searchString)}&page=${
+      this.currentPage
+    }`;
```

Comment:
So we're only going to use the search endpoint? Neat, I'll wire up the correct serializer.